scipy/optimize/tests/test_optimize.py

```diff
@@ -1600,11 +1601,17 @@ def h(x):
         if new_cb_interface == 1:
             def callback_interface(*, intermediate_result):
                 assert intermediate_result.fun == f(intermediate_result.x)
+                if method in MINIMIZE_METHODS_DERIV:
+                    assert intermediate_result.jac == g(intermediate_result.x)
+                    assert intermediate_result.hess == h(intermediate_result.x)
```

Comment:
These are arrays, right? 

```suggestion
                if method in MINIMIZE_METHODS_JAC:
                    assert_allclose(intermediate_result.jac, g(intermediate_result.x))
                if method in MINIMIZE_METHODS_HESS:
                    assert_allclose(intermediate_result.hess, h(intermediate_result.x))
```

Reply:
Sure. I must have been on something when writing this...